lib/mayaUsd/ufe/ProxyShapeHierarchy.cpp

```diff
@@ -135,12 +135,17 @@ Ufe::SceneItem::Ptr ProxyShapeHierarchy::sceneItem() const { return fItem; }
 
 bool ProxyShapeHierarchy::hasChildren() const
 {
+    // Return children of the USD root.
+    const UsdPrim& rootPrim = getUsdRootPrim();
+    if (!rootPrim.IsValid())
+        return false;
+
     // We have an extra logic in createUFEChildList to remap and filter
     // prims. Going this direction is more costly, but easier to maintain.
     //
     // I don't have data that proves we need to worry about performance in here,
     // so going after maintainability.
-    return !children().empty();
+    return !createUFEChildList(getUSDFilteredChildren(rootPrim), false /*filterInactive*/).empty();
```

Comment:
I'm confused by this change.  Now hasChildren() is no longer a convenience for !children().empty(), and is creating a different filtered children list than what children() is doing.  Why is this correct?

Reply:
It is now a convenience for filteredChildren(0 with inactive prim. Unlike children(), which has an alternative function taking a filter that allows specifying to filter deactivated children or not, hasChildren() has a single version. So hasChildren has to choose to report deactivated children or not. The old choice was causing problems because code relied on it even when showing deactivated prims.

If we don't fix it here, then all Maya versions will still have bad outliner behaviour. Changing it fixes the behaviour in all Maya versions that use UFE v2 when they use the latest MayaUSD plugin.